http/action/src/main/java/io/knotx/databridge/http/action/HttpAction.java

```diff
@@ -125,17 +125,60 @@ public void apply(FragmentContext fragmentContext,
   private Single<FragmentResult> process(FragmentContext fragmentContext,
       ActionLogger actionLogger) {
     return Single.just(fragmentContext)
-        .map((ctx) -> buildRequest(ctx, actionLogger))
+        .map(this::createEndpointRequest)
+        .doOnSuccess(request -> logRequest(actionLogger, request))
         .flatMap(
-            request -> callEndpoint(request)
-                .doOnSuccess(resp -> logResponse(request, resp))
+            request -> invokeEndpoint(request)
+                .doOnSuccess(resp -> {
+                  logResponse(request, new HttpResponseData(null, String.valueOf(resp.statusCode()),
+                      resp.statusMessage(), resp.headers(), resp.trailers()), actionLogger);
+                })
+                .doOnError(throwable -> logErrorAndRequest(actionLogger, throwable, request))
                 .map(EndpointResponse::fromHttpResponse)
                 .onErrorReturn(this::handleTimeout)
                 .map(resp -> Pair.of(request, resp)))
-        .flatMap(pair -> getFragmentResult(fragmentContext, pair.getLeft(), pair.getRight(),
+        .flatMap(pair -> createFragmentResult(fragmentContext, pair.getLeft(), pair.getRight(),
             actionLogger));
   }
 
+  private Single<HttpResponse<Buffer>> invokeEndpoint(EndpointRequest request) {
+    return Single.just(request)
+        .map(this::createHttpRequest)
+        .doOnSuccess(this::addPredicates)
+        .flatMap(HttpRequest::rxSend);
+  }
+
+  private void addPredicates(HttpRequest<Buffer> request) {
+    if (isJsonPredicate) {
+      request.expect(io.vertx.reactivex.ext.web.client.predicate.ResponsePredicate
+          .newInstance(IS_JSON_RESPONSE));
+    }
+    attachResponsePredicatesToRequest(request,
+        httpActionOptions.getResponseOptions().getPredicates());
+  }
+
+  private void logRequest(ActionLogger actionLogger, EndpointRequest request) {
+    JsonObject headers = new JsonObject();
+    request.getHeaders().entries().forEach(e -> headers.put(e.getKey(), e.getValue()));
+    actionLogger.info(REQUEST, new JsonObject().put("path", request.getPath())
```

Comment:
we log only headers, but I think it might be useful to log params as well, do we do it somewhere and I don't see it? :D

Reply:
we are logging here path and request, query strings are contained already in the path, therefore we don't have to log them, moreover, there is no possibility to do that because EndpointRequest class contains only path and headers fields, nothing more